### PR TITLE
[CA-122] README.md update

### PR DESCRIPTION
**Pull request by tejas-hosamani**
Relatively detailed Documentation for new contributors.

### CODE REVIEW

**Comment by tejas-hosamani** [comment]
# UPDATED

[License: MIT](https://opensource.org/licenses/MIT) GitHub package.json version Travis (.org) branch Libraries.io dependency status for GitHub repo GitHub contributors GitHub commit activity

# Coding Garden Community App API

This repository contains the source files and documentation for the API of the
Coding Garden Community App. For more information about the Community App
please visit the [App Wiki](https://github.com/CodingGardenCommunity/app-wiki/wiki).

## Running the API locally

> **NOTE:** If you find yourself in some trouble going through this, reach out to us directly on our [Discord server](https://discord.gg/bPBuk3N).

### Prerequisites

1. **NodeJS:** <br>
   Please install [NodeJS >= 10.15.0](https://nodejs.org/en/download/). If you already have it, you're good to go.

1. **Yarn:** <br>
   Visit [Yarn download page](https://yarnpkg.com/en/docs/install#windows-stable). Select your Operating system and follow the instructions. It's as easy as eating a 🍰.

1. **EditorConfig:** <br>
   Please visit [EditorConfig](https://editorconfig.org/) -> `Download a Plugin` section and scroll through to see if you need to install an additional Plugin/Extension for your code editor or IDE. If your IDE needs one, you should be able to find a link to that plugin/extension on that page.

   This prerequisite is directly related to: [`.editorconfig`](https://github.com/CodingGardenCommunity/app-backend/blob/develop/.editorconfig) in the root directory of this project.

   **_More About EditorConfig:_** <br>
   EditorConfig helps maintain consistent coding styles for multiple developers working on the same project across various editors and IDEs. The EditorConfig project consists of a file format for defining coding styles and a collection of text editor plugins that enable editors to read the file format and adhere to defined styles. EditorConfig files are easily readable and they work nicely with version control systems.

---

Once you have the [Prerequisites](#prerequisites) covered:

1. [Clone](https://help.github.com/articles/cloning-a-repository/) this repository from GitHub onto your local computer.

   ```sh
   $ git clone https://github.com/CodingGardenCommunity/app-backend.git
   ```

1. Navigate into the project folder and install all of its necessary dependencies with Yarn.

   ```sh
   $ cd app-backend
   $ yarn install
   ```

1. Install MongoDB and make sure it's running

   - For Mac OSX with [homebrew](http://brew.sh/): `brew install mongodb` then `brew services start mongodb`
   - For Windows and Linux: [MongoDB Installation](https://docs.mongodb.com/manual/installation/)

1. Make a copy of `.env.sample` and rename it to `.env`. You can do so with this simple command:

   > **NOTE:** If you are using Windows Command Prompt, you need to use `copy` instead of `cp`. <br>

   ```sh
   $ cp .env.example .env
   ```

   You don't need to change any values in `.env` file. The default values work well for development purposes.

1. Once you have MongoDB and `.env` file ready, seed the local database by running:

   ```sh
   yarn run seed
   ```

1. To make sure everything is setup properly, run tests.

   ```sh
   yarn run test
   ```

   If all tests pass, we can safely conclude that setup is complete and its working as expected. 🙌 Wooh!! <br>
   If not, don't worry. We are together on this mission!! Reach out to us on our [Discord server](https://discord.gg/bPBuk3N).

1. Once that's done, tap your back. You are ready to start contributing 😃 <br>
   You can run -

   ```sh
   yarn run dev
   ```

   to start the server.

You can now visit <http://localhost:3000/> to view the APIs.

Further, checkout [package.json](https://github.com/CodingGardenCommunity/app-backend/blob/develop/package.json) file to learn about (more) available scrips/commands.

Happy coding! 🥂

**Comment by tejas-hosamani** [comment]
[As a Possible Contributor, I would like to have an up-to-date README.md on app-backend](https://trello.com/c/ZAmqemEW/122-as-a-possible-contributor-i-would-like-to-have-an-up-to-date-readmemd-on-app-backend)

File: README.md [hunk 1]
```diff
@@ -3,45 +3,91 @@
 # Coding Garden Community App API
 
 This repository contains the source files and documentation for the API of the
-Coding Garden Community App. For general information about the Community App
+Coding Garden Community App. For more information about the Community App
 please visit the [App Wiki](https://github.com/CodingGardenCommunity/app-wiki/wiki).
 
 ## Running the API locally
 
+> **NOTE:** If you find yourself in some trouble going through this, reach out to us on our [Discord server](https://github.com/CodingGardenCommunity/app-wiki/wiki/Getting-Involved).
```
Comment (spiray) [inline README.md]: Should this link directly to `https://discord.gg/bPBuk3N`?

File: README.md [hunk 2]
```diff
@@ -3,45 +3,91 @@
...
-1.13.0](https://yarnpkg.com/en/docs/install).
+   ```
+   $ git clone https://github.com/CodingGardenCommunity/app-backend.git
+   ```
 
-After setting up both tools you can install all dependencies by cloning this
-project and running `yarn install` in the local project's root directory.
-Starting the API can then be done by issuing this command: `yarn run dev`. You
-can now visit <http://localhost:3000/> to view the API.
+1. Navigate into the project folder and install all of its necessary dependencies with NPM or Yarn.
```
Comment (spiray) [inline README.md]: This should only be done with yarn as we are committing the lock file.

File: README.md [hunk 3]
```diff
@@ -3,45 +3,91 @@
...
 
-### Via Docker
+   ```
+   $ cd app-backend
+   $ npm install // or yarn install
```
Comment (spiray) [inline README.md]: See previous comment.

File: README.md [hunk 4]
```diff
@@ -3,45 +3,91 @@
...
+   ```
 
-Copy the `.env.sample` file and update accordingly:
+   You don't need to change any values in `.env` file. The default values work well for development purposes.
 
-```sh
-cp .env.sample .env
-```
-Fill the database with data:
+1. Once you have MongoDB and `.env` file ready, populate some data in your local Database to work with
```
Comment (spiray) [inline README.md]: Add "seed the local database by running." or something like that.

File: README.md [hunk 5]
```diff
@@ -3,45 +3,91 @@
...
+   npm run seed
+   ```
+
+1. To make sure everything is setup properly, run tests.
+
+   ```sh
+   npm run test
+   ```
+
+   If all tests pass(they will), we can safely conclude that setup is complete and its working as expected. Wooh!!
```
Comment (spiray) [inline README.md]: Add a space between pass and (.

File: README.md [hunk 6]
```diff
@@ -3,45 +3,91 @@
...
+
+1. Once that's done, tap your back. You are ready to start contributing :)
```
Comment (spiray) [inline README.md]: Change :) to 🙂Add, You can run then have npm run dev on the next line and then right to start the server.

**Review by spiray — CHANGES_REQUESTED** [review]
Thanks @CodingCreate101!

File: README.md (hunk 1 above)
Comment (tejas-hosamani) [reply README.md]: Yes. But I thought, adding that link kinda introduces them to list of other important things.. that they might wanna know..

File: README.md (hunk 2 above)
Comment (tejas-hosamani) [reply README.md]: Yes, I realized that today. I'll update that.

File: README.md (hunk 1 above)
Comment (spiray) [reply README.md]: That makes sense but the preceding verbiage makes it sound like you are linking them directly to the server. You can change it to "here is some info on how to get in touch with us." and use here as a link.

File: README.md (hunk 6 above)
Comment (tejas-hosamani) [reply README.md]: > Add, You can run then have npm run dev on the next line and then right to start the server.

What did you mean? I didn't understand; please explain.

Comment (spiray) [reply README.md]: 7. Once that's done, tab your back. You are ready to start contributing🙂You can run -
```sh
npm run dev
```
to start the server. You can now visit http://localhost:3000/ to view the APIs.

Something like that ^

File: README.md (hunk 1 above)
Comment (tejas-hosamani) [reply README.md]: Yeah, you are right. I 'll change the way it's described. Thanks! 😃

File: README.md [hunk 7]
```diff
@@ -3,45 +3,87 @@
...
+   to start the server.
 
-Copy the `.env.sample` file and update accordingly:
+You can now visit <http://localhost:3000/> to view the APIs.
 
-```sh
-cp .env.sample .env
-```
-Fill the database with data:
+Further, checkout [package.json](https://github.com/CodingGardenCommunity/app-backend/blob/develop/package.json) file to learn about (more) available scrips/commands.
```
Comment (spiray) [inline README.md]: scrips -> scripts

File: README.md [hunk 8]
```diff
@@ -3,45 +3,87 @@
...
+> **NOTE:** If you find yourself in some trouble going through this, reach out to us directly on our [Discord server](https://discord.gg/bPBuk3N).
 
 ### Prerequisites
 
-Go to [EditorConfig](https://editorconfig.org/#download) and see if you need to install a Plugin/Extension for your editor.
+1. **NodeJS:** <br>
+   Please install [NodeJS >= 10.15.0](https://nodejs.org/en/download/). If you already have it, you're good to go.
+
+1. **Yarn:** <br>
+   Visit [Yarn download page](https://yarnpkg.com/en/docs/install#windows-stable). Select your Operating system and follow the instructions. It's as easy as eating a 🍰.
```
Comment (spiray) [inline README.md]: The #windows-stable should be removed from the URL.

**Review by ewenjo — DISMISSED** [review]

File: README.md [hunk 9]
```diff
@@ -3,45 +3,87 @@
...
-1.13.0](https://yarnpkg.com/en/docs/install).
+   ```sh
+   yarn run test
+   ```
 
-After setting up both tools you can install all dependencies by cloning this
-project and running `yarn install` in the local project's root directory.
-Starting the API can then be done by issuing this command: `yarn run dev`. You
-can now visit <http://localhost:3000/> to view the API.
+   If all tests pass (they will), we can safely conclude that setup is complete and its working as expected. 🙌 Wooh!! <br>
```
Comment (spiray) [inline README.md]: Do we want to say "they will!"?

**Review by spiray — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (tejas-hosamani) [reply README.md]: 😜 I am imagining a casual tone, like.. they will.. don't worry..

File: README.md (hunk 7 above)
Comment (tejas-hosamani) [reply README.md]: Man!! How did you find that! I'll be more careful next time.

File: README.md (hunk 9 above)
Comment (spiray) [reply README.md]: I know, but in a way it can discourage contributors if they don't. It doesn't really matter

Comment (tejas-hosamani) [reply README.md]: Hmm.. that's a good point. I'll remove that.

**Review by spiray — APPROVED** [review]
Looks good!

**Review by ewenjo — APPROVED** [review]

**Pull request merged**